server.js

```diff
@@ -465,6 +374,369 @@ io.on('connection', (socket) => {
+                execution();
+            } else if (fPols == 6) {
+                fWins++;
+                io.emit('sh-game-finished', ('Six ' + nameOfFascists + ' policies enacted. ' + nameOfFascists + 's win.'));
+                gameOver = true;
+            } else {
+                nextRound('');
+            }
+        } else {
+            if ((lPols == 5) && (bot.includes(president))) {
+                lWins++;
+                io.emit('sh-game-finished', ('Five ' + nameOfLiberals + ' policies enacted. ' + nameOfLiberals + 's win.'));
+                gameOver = true;
+            } else if (bots.includes(president)) {
+                  nextRound('');
+            }
+        }
+    }
+}
+
+function policyPeek() {
+    fCardCount = 0;
+    for (var i = 0; i < topThreePolicies.length; i++) {
+        if(topThreePolicies[i]) {
+            fCardCount++;
+        }
+    }
+    console.log("Policy peek, " + fCardCount + " fascist cards.");
+    if (((presIndex == shPlayers.length + 1) && (fascists.includes(shPlayers[0]))) || (fascists.includes(shPlayers[presIndex +1]))) {
+        policyLie();
+    } else {
+        policyTruth();
+    }
+    nextRound('');
+}
+
+function policyLie() {
+    console.log("Fascist president. Lying...");
+    if (fCardCount == 3) {
+        peekResult = 3;
+    } else {
+        peekResult = fCardCount + 1;
+    }
+    peekReport();
+}
+
+function policyTruth() {
+    console.log("Liberal president. Reporting...");
+    peekResult = fCardCount;
+    peekReport();
+}
+
+function peekReport() {
+    var message = "";
+    if (peekResult == 3) {
+        message = "There are three " + nameOfFascists + " policies and no " + nameOfLiberals + " policies."
+    } else if (peekResult == 2) {
+        message = "There are two " + nameOfFascists + " policies and one " + nameOfLiberals + " policy."
+    } else if (peekResult == 1) {
+        message = "There is one " + nameOfFascists + " policy and two " + nameOfLiberals + " policies."
+    } else {
+        message = "There are no " + nameOfFascists + " policies and three " + nameOfLiberals + " policies."
+    }
+    var report = {
+        user: president,
+        message: message
+    };
+    io.sockets.emit('sh-message', report);
+    console.log(president + ': ' + report.message);
+}
+
+function execution() {
+    if (fascists.includes(president)) {
+        console.log("Fascist president. Executing liberal...");
+        target = liberals[Math.floor(Math.random() * (liberals.length))];
+    } else {
+        console.log("Liberal president. Executing a random player...");
+        target = shPlayers[Math.floor(Math.random() * (shPlayers.length))];
+    }
+    killPlayer(target);
+    console.log("Executing " + target);
+    if (target == hitler) {
+        lWins++;
+        io.emit('sh-game-finished', nameOfHitler + " was killed! " + nameOfLiberals + "s win!");
+        gameOver = true;
+    } else {
+        nextRound('');
+    }
+}
```

Comment:
the president executes if it is a bot, a random liberal if it is a fascist president and a random player for a liberal president